component/repo-server.jsonnet

```
@@ -10,24 +10,32 @@ local deployment = std.parseJson(kap.yaml_load('argocd/manifests/' + params.git_
 local service = std.parseJson(kap.yaml_load('argocd/manifests/' + params.git_tag + '/repo-server/argocd-repo-server-service.yaml'));
 local vault_agent_config = kube.ConfigMap('vault-agent-config') {
   data: {
-    'vault-agent-config.hcl': |||
-      exit_after_auth = false
-
-      auto_auth {
-          method "kubernetes" {
-              config = {
-                  role = "%s"
-                  token_path = "/var/run/secrets/syn/token"
-              }
-          }
-          sink "file" {
-              config = {
-                  path = "/home/vault/.vault-token"
-                  mode = 0644
-              }
-          }
-      }
-    ||| % inv.parameters.secret_management.vault_role,
+    'vault-agent-config.json': std.manifestJson({
+      exit_after_auth: false,
+      [if std.objectHas(inv.parameters.secret_management, 'vault_mount_path') then 'mount_path']: inv.parameters.secret_management.vault_mount_path,
```

Using `parameters.secret_management.vault_mount_path` could be confusing since we also configure the KV store mount path in `parameters.kapitan.secrets.mount` (not based on a field of `parameters.secret_management` currently).

I'd prefer something like `parameters.secret_management.vault_auth_mount_path`, to make it clear which mount path we're configuring.

```suggestion
      [if std.objectHas(inv.parameters.secret_management, 'vault_auth_mount_path') then 'mount_path']: inv.parameters.secret_management.vault_auth_mount_path,
```